### PR TITLE
[PL-44116] PL 1.21.5 release notes

### PR DESCRIPTION
**Pull request by saifeemustafaqharness**
[PL-44116](https://harness.atlassian.net/browse/PL-44116) PL 1.21.5

https://harness.atlassian.net/wiki/spaces/PD/pages/21609677152/NG+Release+SAAS+January+15th+2024+for+1.21.x-+Build

For reviewers, a preview is available:

https://65aad3538e420615d47bc820--harness-developer.netlify.app/release-notes/platform#version-121x-

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [x] Tested Locally
- [ ] *Optional* Screen Shoot. 

[PL-44116]: https://harness.atlassian.net/browse/PL-44116?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: release-notes/platform.md [hunk 1]
```diff
@@ -82,6 +82,16 @@ The following deprecated API endpoints will no longer be supported:
...
+### Version 1.21.x <!--  January xx, 2024 -->
+
+#### Fixed issues
+
+- Tooltips in the left-hand navigation were incorrectly displaying behind the stage execution details panel, now, tooltip will be clearly visible on the Execution Page. (PL-43993)
+- Fixed the ACL list roles API to correctly display `HarnessManaged`, `CreatedAt`, and `LastModifiedAt` date fields, ensuring accurate role management data in responses. (PL-43952)
+- MultiSelect dropdowns in the Harness NG UI app would "reset-to-top" after each selection. Now, fixed across all MultiSelect dropdowns unless explicitly specified by the user. (PL-43925)
+- When editing user group data, member data was not added as expected. Now, the User Group data related to the User Group Members is not lost when the User Group is updated. (PL-43855, ZD-55944)
+- Fixed an issue where searching for user groups containing special characters resulted in a 500 error due to invalid regex patterns in the search term. Now, the `usergroup` list API validates regex patterns and provides a clear error message for invalid search terms. (PL-43761)
+
```
Comment (brian-f-harness) [inline release-notes/platform.md:94]:
```suggestion
[PL-43333](https://harness.atlassian.net/browse/PL-43333) is also on the list.
```

File: release-notes/platform.md [hunk 2]
```diff
@@ -82,6 +82,16 @@ The following deprecated API endpoints will no longer be supported:
 
 ## January 2024
 
+### Version 1.21.x <!--  January xx, 2024 -->
+
+#### Fixed issues
+
+- Tooltips in the left-hand navigation were incorrectly displaying behind the stage execution details panel, now, tooltip will be clearly visible on the Execution Page. (PL-43993)
+- Fixed the ACL list roles API to correctly display `HarnessManaged`, `CreatedAt`, and `LastModifiedAt` date fields, ensuring accurate role management data in responses. (PL-43952)
+- MultiSelect dropdowns in the Harness NG UI app would "reset-to-top" after each selection. Now, fixed across all MultiSelect dropdowns unless explicitly specified by the user. (PL-43925)
```
Comment (brian-f-harness) [inline release-notes/platform.md:91]:
```suggestion
- Multi-select dropdowns would reset to the top after each selection. This issue is fixed for all multi-select dropdown lists unless explicitly specified by the user. (PL-43925)
```

File: release-notes/platform.md [hunk 3]
```diff
@@ -82,6 +82,16 @@ The following deprecated API endpoints will no longer be supported:
...
+- When editing user group data, member data was not added as expected. Now, the User Group data related to the User Group Members is not lost when the User Group is updated. (PL-43855, ZD-55944)
```
Comment (brian-f-harness) [inline release-notes/platform.md:92]:
```suggestion
- When editing user group data, member data was not added as expected. Now, the user group data related to the user group members is not lost when the user group is updated. (PL-43855, ZD-55944)
```

File: release-notes/platform.md [hunk 4]
```diff
@@ -82,6 +82,17 @@ The following deprecated API endpoints will no longer be supported:
...
+- Fixed the ACL list roles API to correctly display `HarnessManaged`, `CreatedAt`, and `LastModifiedAt` date fields, ensuring accurate role management data in responses. (PL-43952)
+- Multi-select dropdowns would reset to the top after each selection. This issue is fixed for all multi-select dropdowns unless explicitly specified by the user. (PL-43925)
+- When editing user group data, member data was not added as expected. Now, the user group data related to the user group members is not lost when the user group is updated. (PL-43855, ZD-55944)  
+- Fixed an issue where searching for user groups containing special characters resulted in a 500 error due to invalid regex patterns in the search term. Now, the `usergroup` list API validates regex patterns and provides a clear error message for invalid search terms. (PL-43761)
+- The Azure endpoints were not being set based on the Azure environment selected. This led to Azure connector working correctly only for Azure public cloud and not for other variants of Azure cloud (like Azure Gov, Azure China etc.). Now the correct Azure resource manager endpoint we will be choosen based on the environment selected in the connector.(PL-43333)
```
Comment (brian-f-harness) [inline release-notes/platform.md:94]:
```suggestion
- The Azure endpoints were not being set based on the Azure environment selected. This led to Azure connectors working correctly only for Azure public cloud and not for other variants of Azure cloud (like Azure Gov, Azure China, and so on). Now, the correct Azure resource manager endpoint will be chosen based on the environment selected in the connector. (PL-43333, ZD-54717)
```

File: release-notes/platform.md [hunk 5]
```diff
@@ -82,6 +82,17 @@ The following deprecated API endpoints will no longer be supported:
 
 ## January 2024
 
+### Version 1.21.x <!--  January xx, 2024 -->
+
+#### Fixed issues
+
+- Tooltips in the left navigation were incorrectly displayed behind the stage execution details panel. Now, tooltips are visible on the Execution page. (PL-43993)
+- Fixed the ACL list roles API to correctly display `HarnessManaged`, `CreatedAt`, and `LastModifiedAt` date fields, ensuring accurate role management data in responses. (PL-43952)
+- Multi-select dropdowns would reset to the top after each selection. This issue is fixed for all multi-select dropdowns unless explicitly specified by the user. (PL-43925)
```
Comment (brian-f-harness) [inline release-notes/platform.md:91]: Looks like PL-43925 is no longer on the list.

Comment (saifeemustafaqharness) [reply release-notes/platform.md:91]: I do see it on the list. Visible on https://harness.atlassian.net/wiki/spaces/PD/pages/21609677152/NG+Release+SAAS+January+15th+2024+for+1.21.x-+Build

Comment (brian-f-harness) [reply release-notes/platform.md:91]: > I do see it on the list. Visible on https://harness.atlassian.net/wiki/spaces/PD/pages/21609677152/NG+Release+SAAS+January+15th+2024+for+1.21.x-+Build

Thanks @saifeemustafaqharness. It's back on the list today but wasn't there yesterday.

image

**Review by brian-f-harness — APPROVED** [review]
LGTM

**Review by brian-f-harness — DISMISSED** [review]

**Review by pratmit — APPROVED** [review]
LGTM

**Comment by saifeemustafaqharness** [comment]
Hi @nataraja2018 and @harishkammanahalli 

Requesting your review and approval on this. Thanks

**Comment by saifeemustafaqharness** [comment]
> Hi @nataraja2018 and @harishkammanahalli
> 
> Requesting your review and approval on this. Thanks

No action request required. Thanks

**Review by brian-f-harness — APPROVED** [review]

**Pull request merged**